### PR TITLE
fix #87: wait for fresh chain 15x faster

### LINKED ISSUE #87: downloading fresh chain from network takes too long even when everything is working correctly
This has been mentioned in #1.
Some rough metrics, calculated on my head and without using a timing device:
- Chain download speed "feels like" about 10 blocks per second or less
- That translates to over 1 minute for each day that the chain is old, to catch up
- The latest chain.zip is from mid-May
- So we're looking at 1-2 hours before a fresh install can start mining.

### PR DESCRIPTION
**Pull request by atramos**
Implemented a profiler and then optimized the slow parts.

### CODE REVIEW

**Review by lk16 — CHANGES_REQUESTED** [review]
The faster chain loading from peers is working, that is very nice.

However I notice that it wants to download the chain from peers again when I run the script for a second time.

The chain folder does contain all the blocks until 97100. 
At the time of writing the chain height is 97196, so if I understand the comments correctly, this is expected behaviour.

I also cloned your branch in another folder and tried again, just to make sure that my dev folder is not broken somehow. But I got the same results again.

**Comment by atramos** [comment]
Thanks for the feedback. I've switched this PR back to Draft until the comments are addressed.

File: skepticoin/networking/remote_peer.py
Comment (atramos) [reply skepticoin/networking/remote_peer.py:466]: > However I notice that it wants to download the chain from peers again when I run the script for a second time.

I was able to reproduce this. It's fixed in c54ff65900f48e98854f0c6a6645bb05464372cf

Comment (atramos) [reply skepticoin/networking/remote_peer.py:466]: > If we only want to catch block validation issues here and nothing else, this `try` block should not have anything else.
> I suggest moving the rest of the `try` block after the exception handling.

Fixed in d7a48dd1d10679c3cf6b1ed0e4cf20294fe669db

**Comment by atramos** [comment]
As part of the fixes in c54ff65900f48e98854f0c6a6645bb05464372cf, this PR also makes the `chain.cache` the primary block store and phases out the use of a single-file for every block in the chain.

File: skepticoin/scripts/utils.py
Comment (lk16) [inline skepticoin/scripts/utils.py:109]: Time to implement some proper exception handling, see #91. Not in this PR of course.

**Review by lk16 — APPROVED** [review]
I'll approve this, but if you merge this without changing the testcases, please create a new issue for it.

File: skepticoin/scripts/utils.py
Comment (atramos) [reply skepticoin/scripts/utils.py:92]: Created issue #93.

File: skepticoin/scripts/utils.py
Comment (atramos) [reply skepticoin/scripts/utils.py:36]: I'll rename create_chain_dir to check_chain_dir before merging this PR. I'll create an issue about deleting the obsolete files.

File: skepticoin/scripts/utils.py
Comment (atramos) [reply skepticoin/scripts/utils.py:109]: I agree!

**Pull request merged**
